client/modules/page/components/PageEditor.js

```diff
@@ -115,15 +115,19 @@ class PageEditor extends React.Component {
                 />
               </div>
             }
-            <QuillBodyToolbar placeholders={bodyPlaceholders} type={type} />
-            <ReactQuill
-              theme="snow"
-              bounds=".page-editor"
-              value={page.body}
-              onChange={this.handleEditorChange(editorTypes.BODY)}
-              modules={getQuillBodyModules(type)}
-              ref={this.getEditorInstance(editorTypes.BODY)}
-            />
+            {page.body &&
```

Comment:
I had no content on my Donor page so `page.body` was an empty string `""` which caused the editor not to show.  If I changed `page.body` to `page.body !== undefined` this fixed it.

Reply:
Ah, thaks kenjiO, good catch! 

Does this solution seem kind of "hacky" to you as far as fix? That's my main concern. I can look more into the way the component receives props and see if there's a way to make that happen in a way that page.body doesn't ever end up being undefined or "" so we wouldn't have to check at all, if you think that'd be better.

Reply:
Looking into this more: it seems like the problem might be more in the parent EditPages component's `componentWillReceiveProps`. 

<img width="599" alt="screen shot 2017-11-11 at 7 56 28 am" src="https://user-images.githubusercontent.com/18345212/32690222-dc39b544-c6b8-11e7-8ba5-5af9c1a2d0f5.png">

It seems like we're getting `selectedPage` out of sync with `pages` when we grab the new pages in EditPages. We ask PageEditor to grab a `selectedPage` with an identifier that doesn't yet exist in its page array.

In EditPages's `componentWillReceiveProps`, is there a way to make nextProps.loadPage() only call `setState({selectedPage})` after the new pages are loaded?

Does this seem like the right approach? We could avoid the intermediate state where selectedPage is null and ensure that we're only passing PageEditor a selectedPage with an identifier that we know for sure will exist.

Reply:
Great job on this, and nice explanations! That works so I think we should add it, can just do it on line 101 though - `{page && page !== undefined &&` and not need the extra div.

I think you're right that it's maybe not the best way and it would be nice to figure out the root cause. I'm pretty stumped, but that error's coming from the toolbar of the body editor and if you don't try to change its id (just call it 'body-toolbar' say in PageEditor's getQuillBodyModules and QuillBodyToolbar) it goes away, but the placeholders and buttons don't update.